pgjdbc/src/main/java/org/postgresql/core/Parser.java

```diff
@@ -1054,37 +1054,20 @@ private static int parseSql(char[] p_sql, int i, StringBuilder newsql, boolean s
             break;
           } else if (c == '{') { // start of an escape code?
             if (i + 1 < len) {
-              char next = p_sql[i + 1];
-              char nextnext = (i + 2 < len) ? p_sql[i + 2] : '\0';
-              if (next == 'd' || next == 'D') {
-                state = SqlParseState.ESC_TIMEDATE;
-                i++;
-                newsql.append("DATE ");
-                break;
-              } else if (next == 't' || next == 'T') {
-                state = SqlParseState.ESC_TIMEDATE;
-                if (nextnext == 's' || nextnext == 'S') {
-                  // timestamp constant
-                  i += 2;
-                  newsql.append("TIMESTAMP ");
-                } else {
-                  // time constant
-                  i++;
-                  newsql.append("TIME ");
+              SqlParseState[] availableStates = SqlParseState.values();
```

Comment:
afaik it's worth caching.

cached run:
```
Result "escapeFunctionWithDate":
  665.654 ±(99.9%) 6.960 ns/op [Average]
  (min, avg, max) = (625.776, 665.654, 725.398), stdev = 20.521
  CI (99.9%): [658.694, 672.614] (assumes normal distribution)

Secondary result "·gc.alloc.rate":
  2339.752 ±(99.9%) 23.956 MB/sec [Average]
  (min, avg, max) = (2145.052, 2339.752, 2486.556), stdev = 70.636
  CI (99.9%): [2315.796, 2363.709] (assumes normal distribution)

Secondary result "·gc.alloc.rate.norm":
  1632.001 ±(99.9%) 0.001 B/op [Average]
  (min, avg, max) = (1632.000, 1632.001, 1632.008), stdev = 0.002
  CI (99.9%): [1632.000, 1632.002] (assumes normal distribution)

Secondary result "·gc.churn.PS_Eden_Space":
  2328.778 ±(99.9%) 31.127 MB/sec [Average]
  (min, avg, max) = (2081.674, 2328.778, 2558.467), stdev = 91.778
  CI (99.9%): [2297.652, 2359.905] (assumes normal distribution)

Secondary result "·gc.churn.PS_Eden_Space.norm":
  1624.358 ±(99.9%) 14.103 B/op [Average]
  (min, avg, max) = (1541.086, 1624.358, 1727.382), stdev = 41.583
  CI (99.9%): [1610.256, 1638.461] (assumes normal distribution)

Secondary result "·gc.churn.PS_Survivor_Space":
  0.208 ±(99.9%) 0.025 MB/sec [Average]
  (min, avg, max) = (0.093, 0.208, 0.406), stdev = 0.074
  CI (99.9%): [0.183, 0.233] (assumes normal distribution)

Secondary result "·gc.churn.PS_Survivor_Space.norm":
  0.145 ±(99.9%) 0.017 B/op [Average]
  (min, avg, max) = (0.064, 0.145, 0.284), stdev = 0.051
  CI (99.9%): [0.127, 0.162] (assumes normal distribution)

Secondary result "·gc.count":
  1676.000 ±(99.9%) 0.001 counts [Sum]
  (min, avg, max) = (10.000, 16.760, 19.000), stdev = 1.664
  CI (99.9%): [1676.000, 1676.000] (assumes normal distribution)

Secondary result "·gc.time":
  926.000 ±(99.9%) 0.001 ms [Sum]
  (min, avg, max) = (6.000, 9.260, 10.000), stdev = 0.799
  CI (99.9%): [926.000, 926.000] (assumes normal distribution)


# Run complete. Total time: 00:03:25

Benchmark                                                                 Mode  Cnt     Score    Error   Units
EscapeProcessing.escapeFunctionWithDate                                   avgt  100   665.654 ±  6.960   ns/op
EscapeProcessing.escapeFunctionWithDate:·gc.alloc.rate                    avgt  100  2339.752 ± 23.956  MB/sec
EscapeProcessing.escapeFunctionWithDate:·gc.alloc.rate.norm               avgt  100  1632.001 ±  0.001    B/op
EscapeProcessing.escapeFunctionWithDate:·gc.churn.PS_Eden_Space           avgt  100  2328.778 ± 31.127  MB/sec
EscapeProcessing.escapeFunctionWithDate:·gc.churn.PS_Eden_Space.norm      avgt  100  1624.358 ± 14.103    B/op
EscapeProcessing.escapeFunctionWithDate:·gc.churn.PS_Survivor_Space       avgt  100     0.208 ±  0.025  MB/sec
EscapeProcessing.escapeFunctionWithDate:·gc.churn.PS_Survivor_Space.norm  avgt  100     0.145 ±  0.017    B/op
EscapeProcessing.escapeFunctionWithDate:·gc.count                         avgt  100  1676.000           counts
EscapeProcessing.escapeFunctionWithDate:·gc.time                          avgt  100   926.000               ms
```

non-cached, code changed to:
```
  for (int j = 1; j < SqlParseState.values().length; j++) {
    SqlParseState availableState = SqlParseState.values()[j];
 ...
```

non-cached run:
```
Result "escapeFunctionWithDate":
  662.387 ±(99.9%) 7.569 ns/op [Average]
  (min, avg, max) = (608.812, 662.387, 738.422), stdev = 22.318
  CI (99.9%): [654.818, 669.956] (assumes normal distribution)

Secondary result "·gc.alloc.rate":
  2348.322 ±(99.9%) 26.825 MB/sec [Average]
  (min, avg, max) = (2107.321, 2348.322, 2555.984), stdev = 79.093
  CI (99.9%): [2321.498, 2375.147] (assumes normal distribution)

Secondary result "·gc.alloc.rate.norm":
  1629.601 ±(99.9%) 2.454 B/op [Average]
  (min, avg, max) = (1608.000, 1629.601, 1632.008), stdev = 7.236
  CI (99.9%): [1627.147, 1632.055] (assumes normal distribution)

Secondary result "·gc.churn.PS_Eden_Space":
  2344.777 ±(99.9%) 30.917 MB/sec [Average]
  (min, avg, max) = (2093.271, 2344.777, 2585.418), stdev = 91.160
  CI (99.9%): [2313.860, 2375.694] (assumes normal distribution)

Secondary result "·gc.churn.PS_Eden_Space.norm":
  1627.368 ±(99.9%) 14.359 B/op [Average]
  (min, avg, max) = (1498.802, 1627.368, 1726.184), stdev = 42.339
  CI (99.9%): [1613.008, 1641.727] (assumes normal distribution)

Secondary result "·gc.churn.PS_Survivor_Space":
  0.226 ±(99.9%) 0.022 MB/sec [Average]
  (min, avg, max) = (0.062, 0.226, 0.404), stdev = 0.065
  CI (99.9%): [0.204, 0.247] (assumes normal distribution)

Secondary result "·gc.churn.PS_Survivor_Space.norm":
  0.157 ±(99.9%) 0.015 B/op [Average]
  (min, avg, max) = (0.042, 0.157, 0.278), stdev = 0.046
  CI (99.9%): [0.141, 0.172] (assumes normal distribution)

Secondary result "·gc.count":
  1801.000 ±(99.9%) 0.001 counts [Sum]
  (min, avg, max) = (14.000, 18.010, 21.000), stdev = 1.425
  CI (99.9%): [1801.000, 1801.000] (assumes normal distribution)

Secondary result "·gc.time":
  930.000 ±(99.9%) 0.001 ms [Sum]
  (min, avg, max) = (8.000, 9.300, 11.000), stdev = 0.704
  CI (99.9%): [930.000, 930.000] (assumes normal distribution)


# Run complete. Total time: 00:03:24

Benchmark                                                                 Mode  Cnt     Score    Error   Units
EscapeProcessing.escapeFunctionWithDate                                   avgt  100   662.387 ±  7.569   ns/op
EscapeProcessing.escapeFunctionWithDate:·gc.alloc.rate                    avgt  100  2348.322 ± 26.825  MB/sec
EscapeProcessing.escapeFunctionWithDate:·gc.alloc.rate.norm               avgt  100  1629.601 ±  2.454    B/op
EscapeProcessing.escapeFunctionWithDate:·gc.churn.PS_Eden_Space           avgt  100  2344.777 ± 30.917  MB/sec
EscapeProcessing.escapeFunctionWithDate:·gc.churn.PS_Eden_Space.norm      avgt  100  1627.368 ± 14.359    B/op
EscapeProcessing.escapeFunctionWithDate:·gc.churn.PS_Survivor_Space       avgt  100     0.226 ±  0.022  MB/sec
EscapeProcessing.escapeFunctionWithDate:·gc.churn.PS_Survivor_Space.norm  avgt  100     0.157 ±  0.015    B/op
EscapeProcessing.escapeFunctionWithDate:·gc.count                         avgt  100  1801.000           counts
EscapeProcessing.escapeFunctionWithDate:·gc.time                          avgt  100   930.000               ms
```